tcl/linter.tcl

```diff
@@ -320,3 +320,59 @@ proc linter_sql_query_indices {string} {
 
     return [regexp -all -inline -indices $pattern $string]
 }
+
+proc linter_procs_without_proc_comment {string} {
+    #| Get a list procs that don't have a proc comment.
+
+    set procs [list]
+    set commands [linter_tcl_commands $string]
+
+    foreach command $commands {
+        switch -regexp $command {
```

Comment:
I've initially checked the command as a string because `lassign` and `lindex` cannot handle some commands.

A failing case that I came across in a file is a namespace where there's a lappend that has commands as arguments:

```tcl
namespace eval technical_patch {

    set system_columns [list]
    lappend system_columns \
        [dict create \
             table "product" \
             column "product_code" \
             units "" \
             label "Code"] \
        ...
```

```
list element in quotes followed by "]" instead of space
    while executing
"lassign $command command_name arg1 arg2 arg3"
```